tests/test_bookmarks.py

```diff
@@ -0,0 +1,143 @@
+                                        if record.get('action') == 'upsert']
+                first_bookmark_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from sync 1 & 2
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    first_bookmark_value_utc = self.parse_date(first_bookmark_value[replication_key])
+                    second_bookmark_value_utc = self.parse_date(second_bookmark_value[replication_key])
+                    simulated_bookmark_utc = self.parse_date(simulated_states['bookmarks'][stream][replication_key])
+
+                    # verify the syncs sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_value)
+                    self.assertIsNotNone(second_bookmark_value)
+                    
+                    self.assertIsNotNone(first_bookmark_value[replication_key])
+                    self.assertIsNotNone(second_bookmark_value[replication_key])
+
+                    # verify the 2nd bookmark is equal to 1st sync bookmark
+                    self.assertEqual(first_bookmark_value[replication_key],
+                                     second_bookmark_value[replication_key])
+
+                    for record in first_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify 1st sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, first_bookmark_value_utc,
+                                             msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    for record in second_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value_utc, simulated_bookmark_utc,
+                                                msg="Second sync records do not respect the previous bookmark")
+                        # verify the 2nd sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, second_bookmark_value_utc,
+                                             msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    # verify that we get less data in the 2nd sync
```

Comment:
// Verify the number of records in records_1 where sync >= manipulated_start_time matches the number of records in records_2
     records_1_after_manipulated_bookmark = 0
     for record in sync_1_records:
          rec_time = self.parse_date(record.get(expected_replication_key))
          if rec_time >= manipulated_start_time:
               records_1_after_manipulated_bookmark += 1
          self.assertEqual(records_1_after_manipulated_bookmark, len(sync_2_records), msg="...")